File: gbi-scios-frontend/types/project.ts
// types/project.ts

/**
 * Mirrors backend `ProjectRead` (app/schemas/project.py):
 * { id, project_name, client_name, sector, status, created_at, owner_id }
 *
 * The frontend keeps `name`/`client` aliases (mapped in projectService) so
 * existing UI components keep working without a wide refactor.
 *
 * risk_level / compliance_score are intentionally removed — they do NOT exist
 * on the backend ProjectRead (audit finding FE-9).
 */

export type ProjectStatus =
  | "Planning"
  | "Active"
  | "On Hold"
  | "Completed"
  | "Cancelled";

export interface Project {
  id: string;
  // Frontend-friendly aliases (mapped from project_name / client_name).
  name: string;
  client: string;
  sector: string;
  status: ProjectStatus;
  created_at: string;
  owner_id: string;
}

export interface ProjectCreateInput {
  project_name: string;
  client_name: string;
  sector?: string;
  status?: ProjectStatus;
}

export interface ProjectUpdateInput {
  project_name?: string;
  client_name?: string;
  sector?: string;
  status?: ProjectStatus;
}

File: build-a-production-grade-fastapi-backend/app/schemas/project.py
from datetime import datetime
from uuid import UUID

from pydantic import BaseModel, ConfigDict, Field

from app.models.project import ProjectStatus


class ProjectBase(BaseModel):
    project_name: str = Field(min_length=2, max_length=255)
    client_name: str = Field(min_length=2, max_length=255)
    sector: str = Field(min_length=2, max_length=120)
    status: ProjectStatus = ProjectStatus.PLANNING


class ProjectCreate(ProjectBase):
    pass


class ProjectUpdate(BaseModel):
    project_name: str | None = Field(default=None, min_length=2, max_length=255)
    client_name: str | None = Field(default=None, min_length=2, max_length=255)
    sector: str | None = Field(default=None, min_length=2, max_length=120)
    status: ProjectStatus | None = None


class ProjectRead(ProjectBase):
    id: UUID
    created_at: datetime
    owner_id: UUID

    model_config = ConfigDict(from_attributes=True)
